interview-questions.md

```diff
@@ -5,50 +5,81 @@ Answer the following questions. First, without external resources. Challenge you
 
 5. What is React.Fragment and why would you want to use it?
   
   Your answer:
+  Honestly, I'm drawing a blank. I dont think we covered it a whole lot and ontop of that, I dont feel secure in my React knowledge at all. Error 500. Proceeding to reteach myself in my research and extended study of React. 
   
   Researched answer:
+  "A common pattern in React is for a component to return multiple elements. Fragments let you group a list of children without adding extra nodes to the DOM."
+  While my researched study on this produced the above answer straight from React's website, I don't feel like this is really taking hold with me, but thats due to my unease with React. I'm currently taking an extra Udemy and a Codecademy course on React to suppliment my study.  
+  
 
 6. What are three options for creating responsive design?
 
   Your answer:
-
+  1: Media queries.
+  2: Using css class atributes such as flex, and margin. 
+  3: Individual stylesheets that can be loaded upon identification of device type.
+  
   Researched answer:
+  "Responsive website design consists of the following three main components:
+  Flexible layouts - Using a flexible grid to create the website layout that will dynamically resize to any width.
+  Media queries - An extension to media types when targeting and including styles. Media queries allow designers to specify different styles for specific browser and device circumstances.
+  Flexible media - Makes media (images, video and other formats) scalable, by changing the size of the media as the size of the viewport changes."
 
 
 
 ## Looking Ahead: Terms for Next Week
 - Object oriented programming
+  A type of programming that allows developers to define data type of a specific data structure, along with its functions. 
+  Its.. kind of hard to explain in just a sentance or two. 
+
 - irb
+  Interactive Ruby, a program that will show the results of any Ruby statement. You run it in the terminal. 
+
 - Instance variable
+  A type of variable, starting with the @ symbol. You'd use this to store data, or from what I can tell, properties of an object. 
+
 - Ruby hash
+  Similar to an array, a hash is a collection of key and value pairs
+
 - Class relationships
+Ah, boy, so reading on this isnt a simple as a few words here to easily explain what class relationships are. They appear to be a way to define some attributes to link one thing kind of like children of a parent object. For instance, Andee owns a cofee mug with some cool design on them and what they are used for, but he doesnt just own a single coffee mug, he also owns a lot of different coffee mugs with a lot of different attributes, so I'd have to link them all up to the class of Andee and those cups to him.. all with class relationships. 
+
+My brain hurt. 
```

Comment:
this is an excellent analogy! yes! Andee has-many mugs, but a mug is owned by one Andee. And that's their relationship. woo!